internal/jbig2/segments/generic-region.go

```diff
@@ -0,0 +1,1117 @@
+	}
+
+	return g
+}
+
+// Init implements Segmenter interface.
+func (g *GenericRegion) Init(h *Header, r reader.StreamReader) error {
+	g.RegionSegment = NewRegionSegment(r)
+	g.r = r
+	return g.parseHeader()
+}
+
+// GetRegionBitmap gets the bitmap for the generic region segment.
+func (g *GenericRegion) GetRegionBitmap() (bm *bitmap.Bitmap, err error) {
+	if g.Bitmap != nil {
+		return g.Bitmap, nil
+	}
+
+	// Check how the data is encoded.
+	if g.IsMMREncoded {
+		// MMR Decoder Call
+		if g.mmrDecoder == nil {
+			g.mmrDecoder, err = mmr.New(
+				g.r,
+				g.RegionSegment.BitmapWidth, g.RegionSegment.BitmapHeight,
+				g.DataOffset, g.DataLength,
+			)
+			if err != nil {
+				return
+			}
+		}
+
+		// Uncompress the bitmap.
+		g.Bitmap, err = g.mmrDecoder.UncompressMMR()
+		return g.Bitmap, err
+	}
+
+	// Arithmetic decoder process for generic region segments.
+	if err = g.updateOverrideFlags(); err != nil {
+		return
+	}
+
+	// 6.2.5.7 - 1)
+	var ltp int
+	if g.arithDecoder == nil {
+		g.arithDecoder, err = arithmetic.New(g.r)
+		if err != nil {
+			return
+		}
+	}
+	if g.cx == nil {
+		g.cx = arithmetic.NewStats(65536, 1)
+	}
+
+	// 6.2.5.7 - 2)
+	g.Bitmap = bitmap.New(g.RegionSegment.BitmapWidth, g.RegionSegment.BitmapHeight)
+	paddedWidth := int(uint32(g.Bitmap.Width+7) & (^uint32(7)))
+
+	// 6.2.5.7 - 3)
+	for line := 0; line < g.Bitmap.Height; line++ {
+		// 6.2.5.7 - 3 c)
+		if g.IsTPGDon {
+			var temp int
+			temp, err = g.decodeSLTP()
+			if err != nil {
+				return
+			}
+			ltp ^= temp
+		}
+
+		// 6.2.5.7 - 3 d)
+		if ltp == 1 {
+			if line > 0 {
+				if err = g.copyLineAbove(line); err != nil {
+					return
+				}
+			}
+		} else {
+			if err = g.decodeLine(line, g.Bitmap.Width, paddedWidth); err != nil {
+				return
+			}
+		}
+	}
+	return g.Bitmap, nil
+}
+
+// GetRegionInfo implements Regioner interface.
+func (g *GenericRegion) GetRegionInfo() *RegionSegment {
+	return g.RegionSegment
+}
+
+func (g *GenericRegion) parseHeader() (err error) {
+	common.Log.Trace("[GENERIC-REGION] ParsingHeader...")
+	defer func() {
+		if err != nil {
+			common.Log.Trace("[GENERIC-REGION] ParsingHeader Finished with error. %v", err)
+		} else {
+			common.Log.Trace("[GENERIC-REGION] ParsingHeader Finished Succesfully...")
+		}
+	}()
+	var (
+		b    int
+		bits uint64
+	)
+
+	if err = g.RegionSegment.parseHeader(); err != nil {
+		return err
+	}
+
+	// Bit 5-7
+	if _, err = g.r.ReadBits(3); err != nil {
+		return err
+	}
+
+	// Bit 4
+	b, err = g.r.ReadBit()
+	if err != nil {
+		return err
+	}
+	if b == 1 {
+		g.UseExtTemplates = true
+	}
+
+	// Bit 3
+	b, err = g.r.ReadBit()
+	if err != nil {
+		return err
+	}
+	if b == 1 {
+		g.IsTPGDon = true
+	}
+
+	// Bit 1-2
+	bits, err = g.r.ReadBits(2)
+	if err != nil {
+		return err
+	}
+	g.GBTemplate = byte(bits & 0xf)
+
+	// Bit 0
+	b, err = g.r.ReadBit()
+	if err != nil {
+		return err
+	}
+	if b == 1 {
+		g.IsMMREncoded = true
+	}
+
+	if !g.IsMMREncoded {
+		var numberOfGbAt int
```

Comment:
Could initialize `numberOfGbAt` with `1` and remove the else branch. Also inside the if statement, the else branch can be eliminated by assigned `4` to `numberOfGbAt`.